insights/v1alpha1/types_insights.go

```diff
@@ -0,0 +1,247 @@
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+	// state allows you to configure specific gatherer. Valid values are "Enabled", "Disabled" and omitted.
+	// When omitted, this means no opinion and the platform is left to choose a reasonable default.
+	// The current default is Enabled.
+	// +optional
+	State GathererState `json:"state"`
+}
+
+// dataGatherState declares valid gathering state types
+// +kubebuilder:validation:Optional
+// +kubebuilder:validation:Enum=Running;Completed;Failed;Pending
+// +kubebuilder:validation:XValidation:rule="!(oldSelf == 'Running' && self == 'Pending') && !(oldSelf == 'Completed' && self == 'Pending') && !(oldSelf == 'Failed' && self == 'Pending') && !(oldSelf == 'Completed' && self == 'Running') && !(oldSelf == 'Failed' && self == 'Running')", message="state cannot be changed backwards"
+type DataGatherState string
+
+// +kubebuilder:validation:XValidation:rule="(!has(oldSelf.insightsRequestID) || has(self.insightsRequestID)) && (!has(oldSelf.startTime) || has(self.startTime)) && (!has(oldSelf.finishTime) || has(self.finishTime)) && (!has(oldSelf.dataGatherState) || has(self.dataGatherState))",message="cannot remove attributes from status"
+// +kubebuilder:validation:Optional
+type DataGatherStatus struct {
+	// conditions provide details on the status of the gatherer job.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	// +optional
+	Conditions []metav1.Condition `json:"conditions" patchStrategy:"merge" patchMergeKey:"type"`
+	// dataGatherState reflects the current state of the data gathering process.
+	// +optional
+	State DataGatherState `json:"dataGatherState,omitempty"`
+	// gatherers is a list of active gatherers (and their statuses) in the last gathering.
+	// +listType=map
+	// +listMapKey=name
+	// +optional
+	Gatherers []GathererStatus `json:"gatherers,omitempty"`
+	// startTime is the time when Insights data gathering started.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="startTime is immutable once set"
+	// +optional
+	StartTime metav1.Time `json:"startTime,omitempty"`
+	// finishTime is the time when Insights data gathering finished.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="finishTime is immutable once set"
+	// +optional
+	FinishTime metav1.Time `json:"finishTime,omitempty"`
+	// relatedObjects is a list of resources which are useful when debugging or inspecting the data
+	// gathering Pod
+	// +optional
+	RelatedObjects []ObjectReference `json:"relatedObjects,omitempty"`
+	// insightsRequestID is an Insights request ID to track the status of the
+	// Insights analysis (in console.redhat.com processing pipeline) for the corresponding Insights data archive.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="insightsRequestID is immutable once set"
```

Comment:
This might break, you probably want
```suggestion
	// +kubebuilder:validation:XValidation:rule="oldSelf == '' || self == oldSelf",message="insightsRequestID is immutable once set"
```

Reply:
Hm let me experiment it with that more, but it's the example from https://kubernetes.io/blog/2022/09/29/enforce-immutability-using-cel/#immutablility-after-first-modification

Reply:
OK this works for the string attribute, however it's not clear to me whether it's required or not (because it seems working fine without it)

Reply:
as above, I suspect this may break, but I'm willing to loosen once you see it.